### PR TITLE
Remove REAX and MEAM packages

### LINKED ISSUE #1129: [Feature Request] Remove MEAM and REAX packages
## Summary

Both, pair styles `reax` and `meam` were flagged as obsolete for a long time and should be removed after the next stable release.

## Type of Issue

Enhancement

## Detailed Description (Enhancement Suggestion)

Including Fortran libraries complicates the build process, the large static memory allocations for `reax` creates problems when trying to debug or run with valgrind, and reax is not friendly for pre-compiled binaries, since array dimensions may need to be adjusted and the executable recompiled for larger systems. Both packages are incompatible with pair style hybrid. For both there are alternatives written in c++ (`reax/c` and `reax/c/kk`, and `meam/c`). `reax/c` offers more functionality and dynamical memory management (`reax/c/kk` is more robust, though). `meam/c` is a fully compatible re-programming of the Fortran code with performance enhancements (due to avoiding to translate between fortran and c-style data structures and more). Thus `meam/c` is fully backward compatible and we're going to keep support for inputs use the `meam` pair style by making it an (undocumented) alias to `meam/c`. For reax, we'll include a dummy pair style pointing to the last version supporting it or `reax/c`.

Individual steps required:
- [x] make warning about pending removal more drastic and visible
- [x] post a message to lammps-users
- [x] removal of package folders
- [x] removal of library folders
- [x] update of `src/Makefile`
- [x] update of `cmake/CMakelists.txt`
- [x] update of `src/Purge.list`
- [x] update of `src/.gitignore`
- [x] implement dummy pair style class PairObsolete, that will print a meaningful message when obsolete inputs are run (e.g. pointing to the last release containing the package/style) 
- [x] removal or update of pair style and fix style documentation
- [x] removal of all references to REAX or MEAM packages in documentation

## Contingency plan

Users, that insist on using those styles are recommended to use the upcoming stable LAMMPS release and not upgrade, in case they want to keep using REAX or MEAM.

### PR DESCRIPTION
**Pull request by akohlmey**
## Purpose

This pull request fixes #1129 by implementing the changes listed in the issue and removing the MEAM and REAX packages from LAMMPS. The packages were discouraged from use for a while already.

## Author(s)

Axel Kohlmeyer (Temple U)

## Backward Compatibility

Inputs with `pair_style reax` commands will be given a message pointing out the last LAMMPS version that contained this pair style. They need to change the syntax to accommodate the `pair_style reax/c` syntax (and have to have the USER-REAXC package installed). Inputs with `pair_style meam` commands will be redirected transparently to the `pair_style meam/c`, provided the USER-MEAMC package is installed.

## Implementation Notes

For details, please see issue #1129 

## Post Submission Checklist

_Please check the fields below as they are completed_
- [x] The feature or features in this pull request is complete
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] One or more example input decks are included
- [x] The source code follows the LAMMPS formatting guidelines

### CODE REVIEW

File: src/pair_deprecated.cpp [hunk 1]
```diff
@@ -48,6 +48,10 @@ void PairDeprecated::settings(int, char **)
     my_style = hybrid->keywords[hybrid->nstyles];
   }
 
+  if (strcmp(my_style,"reax") == 0) {
+    writemsg(lmp,"\nPair style 'reax' has been removed from LAMMPS "
+                    "after the 12 December 2018 version\n\n",1);
+
```
Comment (rbberger) [inline src/pair_deprecated.cpp]: Missing closing curly bracket `}`

Comment (akohlmey) [reply src/pair_deprecated.cpp]: thanks. done.

**Comment by akohlmey** [comment]
not sure why i had to trigger the rebuild multiple times and only the merge conflict resolve did the trick to have the rebuild work without a failure and also handle the merge conflict. :-1:

**Comment by rbberger** [comment]
@akohlmey applied some new settings to the PR builder. That should avoid this issue.

**Pull request merged**